plasmapy/utils/checks.py

```diff
@@ -102,6 +104,13 @@ def check_quantity(**validations):
         are provided, a `UnitsWarning` will be raised and the inputted
         units will be assumed.
 
+    Notes
+    -----
+    This functionality may end up becoming deprecated due to 
+    noncompliance with the `IEEE 754 standard
+    <https://en.wikipedia.org/wiki/IEEE_754#Exception_handling>`_
+    and in favor of `~astropy.units.quantity_input`.
```

Comment:
It'd be helpful to say the reason why this is in noncompliance with IEEE 754, either in the docstring or in a comment in the function itself.